src/pyobo/sources/uniprot/uniprot.py

```diff
@@ -42,13 +66,71 @@ def iter_terms(version: Optional[str] = None) -> Iterable[Term]:
     """Iterate over UniProt Terms."""
     with open_reader(ensure(version=version)) as reader:
         _ = next(reader)  # header
-        for uniprot_id, name, taxonomy_id, _synonyms, ecs, pubmeds, pdbs in tqdm(
-            reader, desc="Mapping UniProt", unit_scale=True
-        ):
-            term = Term.from_triple(prefix=PREFIX, identifier=uniprot_id, name=name)
-            # TODO add gene encodes from relationship
-            # TODO add description
+        for (
+            uniprot_id,
+            accession,
+            taxonomy_id,
+            _name,  # this field should have the name, but it's a mismatch of random name annotations
+            ecs,
+            pubmeds,
+            pdbs,
+            proteome,
+            gene_id,
+            rhea_curies,
+            go_components,
+            go_functions,
+            go_processes,
+            bindings,
+            description,
+        ) in tqdm(reader, desc="Mapping UniProt", unit_scale=True):
+            term = Term(
+                reference=Reference(prefix=PREFIX, identifier=uniprot_id, name=accession),
+                definition=description or None,
+            )
             term.set_species(taxonomy_id)
+            if gene_id:
+                term.append_relationship(
+                    gene_product_of, Reference(prefix="ncbigene", identifier=gene_id)
+                )
+
+            # TODO add type=Reference(prefix="xsd", identifier="boolean")
+            term.append_property("reviewed", "true")
+
+            for go_process_ref in _parse_go(go_processes):
+                term.append_relationship(participates_in, go_process_ref)
+            for go_function_ref in _parse_go(go_functions):
+                term.append_relationship(enables, go_function_ref)
+            for _go_component_ref in _parse_go(go_components):
+                pass  # TODO what is the right relation?
```

Comment:
@realmarcin the model you gave didn't make an explicit differentiation between go processes, components, and locations. Can you help me find or a better relation between a protein and a cellular component implied by a GO annotation appearing in its UniProt record?